### PR TITLE
Add batch size bytes metric to batch processor

### PR DESCRIPTION
**Pull request by pavolloffay**
**Description:** <Describe what has changed. 

Add batch bytes size metrics to batch processor. 

**Link to tracking Issue:** <Issue number if applicable>

Depends on #1241 
Related to https://github.com/jaegertracing/jaeger/issues/2165#issuecomment-651179655
Related to https://github.com/open-telemetry/opentelemetry-collector/issues/662
Related to https://github.com/open-telemetry/opentelemetry-collector/issues/1139

**Testing:** < Describe what testing was performed and which tests were added.>

**Documentation:** < Describe the documentation added.>

### CODE REVIEW

**Comment by pavolloffay** [comment]
The PR is ready for review.

**Comment by pavolloffay** [comment]
Can somebody restart the `loadtest`?

File: processor/batchprocessor/batch_processor.go [hunk 1]
```diff
@@ -234,6 +235,7 @@ func (bp *batchMetricProcessor) sendItems(measure *stats.Int64Measure) {
 	statsTags := []tag.Mutator{tag.Insert(processor.TagProcessorNameKey, bp.name)}
 	_ = stats.RecordWithTags(context.Background(), statsTags, measure.M(1))
 	_ = stats.RecordWithTags(context.Background(), statsTags, statBatchSendSize.M(int64(bp.batchMetrics.metricData.MetricCount())))
+	_ = stats.RecordWithTags(context.Background(), statsTags, statBatchSendSizeBytes.M(int64(bp.batchMetrics.metricData.Size())))
```
Comment (bogdandrutu) [inline processor/batchprocessor/batch_processor.go:238]: Please combine all Records in one.

Comment (pavolloffay) [reply processor/batchprocessor/batch_processor.go]: done

**Comment by pavolloffay** [comment]
Let's have a look at the benchmark. For a tracedata sample of size 8192 (default batch size) it takes on average 0.5 ms. Compared to span count it is definitely a way more expensive. I have also noticed that for small batches the size operation does not produce any allocations.

```go
func BenchmarkTraceSizeBytes(b *testing.B) {
	td := testdata.GenerateTraceDataManySpansSameResource(8192)
	for n := 0; n < b.N; n++ {
		td.Size()
	}
}

func BenchmarkTraceSizeSpanCount(b *testing.B) {
	td := testdata.GenerateTraceDataManySpansSameResource(8192)
	for n := 0; n < b.N; n++ {
		td.SpanCount()
	}
}
```

```
go test -test.benchtime=3s  -benchmem -cpuprofile profile_cpu.out  -bench=. ./processor/batchprocessor
goos: linux
goarch: amd64
pkg: go.opentelemetry.io/collector/processor/batchprocessor
BenchmarkTraceSizeBytes-8            	    7201	    515344 ns/op	     558 B/op	       7 allocs/op
BenchmarkTraceSizeSpanCount-8        	715425315	         4.68 ns/op	       0 B/op	       0 allocs/op
PASS
ok  	go.opentelemetry.io/collector/processor/batchprocessor	20.311s

```

File: processor/batchprocessor/batch_processor.go [hunk 2]
```diff
@@ -71,11 +73,12 @@ var _ consumer.TraceConsumer = (*batchTraceProcessor)(nil)
 var _ consumer.MetricsConsumer = (*batchMetricProcessor)(nil)
 
 // newBatchTracesProcessor creates a new batch processor that batches traces by size or with timeout
-func newBatchTracesProcessor(params component.ProcessorCreateParams, trace consumer.TraceConsumer, cfg *Config) *batchTraceProcessor {
+func newBatchTracesProcessor(params component.ProcessorCreateParams, trace consumer.TraceConsumer, cfg *Config, telemetryLevel telemetry.Level) *batchTraceProcessor {
```
Comment (pavolloffay) [inline processor/batchprocessor/batch_processor.go:76]: `telemetryLevel` could be added to the `Config` or perhaps `ProcessorCreateParams`?

File: processor/batchprocessor/metrics.go [hunk 3]
```diff
@@ -57,10 +68,21 @@ func MetricViews() []*view.View {
 		Aggregation: view.Distribution(10, 25, 50, 75, 100, 250, 500, 750, 1000, 2000, 3000, 4000, 5000, 6000, 7000, 8000, 9000, 10000, 20000, 30000, 50000, 100000),
 	}
 
+	distributionBatchSendSizeBytesView := &view.View{
+		Name:        statBatchSendSizeBytes.Name(),
+		Measure:     statBatchSendSizeBytes,
+		Description: statBatchSendSizeBytes.Description(),
+		TagKeys:     tagKeys,
```
Comment (pavolloffay) [inline processor/batchprocessor/metrics.go:75]: @bogdandrutu I am not sure if I got this right. 

The `processor.MetricTagKeys(level)` based on the level adds `service` and `processor` tags. Whereas the `processorTagKeys` contains only the `processor` tag. 

Shouldn't we use `tagKeys` for all views?

**Comment by pavolloffay** [comment]
@bogdandrutu PR rebased on your latest refactoring.

File: processor/batchprocessor/metrics.go (hunk 3 above)
Comment (bogdandrutu) [reply processor/batchprocessor/metrics.go:72]: when you record the size you record per service? It does not seem that you do, so you should use `processorTagKeys`

Comment (pavolloffay) [reply processor/batchprocessor/metrics.go]: Done, removed

File: processor/batchprocessor/batch_processor.go (hunk 2 above)
Comment (bogdandrutu) [reply processor/batchprocessor/batch_processor.go]: Please file an issue

**Comment by bogdandrutu** [comment]
There are some failing teats

**Comment by jrcamp** [comment]
check-links CI failure is unrelated to this PR.

File: processor/batchprocessor/batch_processor.go (hunk 1 above)
Comment (pavolloffay) [reply processor/batchprocessor/batch_processor.go]: done https://github.com/open-telemetry/opentelemetry-collector/issues/1413

File: processor/batchprocessor/batch_processor.go (hunk 2 above)
Comment (pavolloffay) [reply processor/batchprocessor/batch_processor.go]: Done https://github.com/open-telemetry/opentelemetry-collector/issues/1413

**Pull request merged**